rpcserver.go

```diff
@@ -1192,3 +1298,51 @@ func marshallSendAssetEvent(
 		return nil, fmt.Errorf("unknown event type: %T", eventInterface)
 	}
 }
+
+// unmarshalScriptKey parses the RPC script key into the native counterpart.
```

Comment:
Double checking; for these keys, we want to accept a pubkey AND an optional key descriptor. For the descriptor, the tweak can be nil since the default is also nil.

For the descriptor, we want to accept a pubkey AND an optional key locator.

Is there any key family sanity checking we should have here?

Also, if the above is correct, maybe `unmarshalScriptKey` should parse the pubkey before the key descriptor so the priority is more clear.

Reply:
Good questions, I guess the code could use a few explanatory comments (which I added).

On the script key level, we allow the descriptor to be nil, which would then indicate "this key is not derived by the wallet that's backing the Taro daemon". So a fully externally derived key.

The tweak being `nil` indicates the usage of BIP 86 as opposed to having a script tree that's being committed to.

On the key descriptor level, we inherit the rules from `lnd`'s APIs: A public key is generally enough to find the corresponding private key (e.g. when signing). But when available the locator information should still be provided, but it's not strictly mandatory.